src/propagation/propagation.jl

```diff
@@ -44,6 +42,26 @@ function scaled_variables(names::String = "δx", c::Vector{T} = fill(1e-6, 6); o
     return dq
 end
 
+# Definition of zero TaylorInterpolant{T, U, 2}
+function zero(::Type{TaylorInterpolant{T, U, 2}}) where {T <: Real, U <: Number}
+    return TaylorInterpolant{T, U, 2}(zero(T), zeros(T, 1), Matrix{Taylor1{U}}(undef, 0, 0))
+end
+
+iszero(x::TaylorInterpolant{T, U, 2}) where {T <: Real, U <: Number} = x == zero(TaylorInterpolant{T, U, 2})
+
+function unsuccessful_propagation(sol::TaylorInterpolant{T, U, 2}, t::T) where {T <: Real, U <: Number}
+
+    if issorted(sol.t)
+        c = sol.t[end] < t
+    elseif issorted(sol.t, rev = true)
+        c = sol.t[end] > t
+    else
+        c = true
+    end
+
+    return iszero(sol) || any(norm.(sol.x, Inf) .> 10) || c 
+end
+
```

Comment:
Since this is a predicate, I would suggest to rename it to something such as `issuccessfulprop` (and changing the logic appropriately)

Reply:
I've changed all `unsuccessful_propagation` to `issuccessfulprop`, as well as the corresponding logic.